sycl/unittests/program_manager/passing_link_and_compile_options.cpp

```diff
@@ -0,0 +1,529 @@
+  auto KernelID_1 = sycl::get_kernel_id<EAMTestKernel1>();
+  auto KernelID_2 = sycl::get_kernel_id<EAMTestKernel2>();
+  sycl::queue Queue{Dev};
+  const sycl::context Ctx = Queue.get_context();
+  sycl::kernel_bundle KernelBundle1 =
+      sycl::get_kernel_bundle<sycl::bundle_state::input>(Ctx, {Dev},
+                                                         {KernelID_1});
+  auto BundleObj1 = sycl::compile(KernelBundle1);
+  sycl::link(BundleObj1);
+  EXPECT_EQ(expected_link_options, current_link_options);
+  EXPECT_EQ(expected_compile_options1, current_compile_options);
+  current_link_options.clear();
+  current_compile_options.clear();
+  sycl::kernel_bundle KernelBundle2 =
+      sycl::get_kernel_bundle<sycl::bundle_state::input>(Ctx, {Dev},
+                                                         {KernelID_2});
+  auto BundleObj2 = sycl::compile(KernelBundle2);
+  sycl::link(BundleObj2);
+  EXPECT_EQ(expected_compile_options2, current_compile_options);
+  EXPECT_EQ(expected_link_options, current_link_options);
+  // Check devices' compile & link options together
+  current_link_options.clear();
+  current_compile_options.clear();
+  static sycl::unittest::PiImageArray<2> DevImageArray[] = {&DevImage1,
+                                                            &DevImage2};
+  sycl::kernel_bundle KernelBundle =
+      sycl::get_kernel_bundle<sycl::bundle_state::input>(
+          Ctx, {Dev}, {KernelID_1, KernelID_2});
+  auto BundleObj = sycl::compile(KernelBundle);
+  sycl::link(BundleObj);
+  EXPECT_EQ("", current_link_options);
+  EXPECT_EQ(expected_compile_options1 + " " + expected_compile_options2,
+            current_compile_options);
+  // empty str + not empty str
+  DevImage1 = generateEAMTestKernel1Image<EAMTestKernel1>(
+      expected_compile_options1_1, expected_link_options);
+  current_link_options.clear();
+  current_compile_options.clear();
+  static sycl::unittest::PiImageArray<2> DevImageArray1[] = {&DevImage1,
+                                                             &DevImage2};
+
+  BundleObj = sycl::compile(KernelBundle);
+  sycl::link(BundleObj);
+  EXPECT_EQ("", current_link_options);
+  EXPECT_EQ(expected_compile_options2, current_compile_options);
+  // not empty str + empty
+  DevImage1 = generateEAMTestKernel1Image<EAMTestKernel1>(
+      expected_compile_options1, expected_link_options);
+  DevImage2 = generateEAMTestKernel1Image<EAMTestKernel2>(
+      expected_compile_options2_1, expected_link_options);
+  current_link_options.clear();
+  current_compile_options.clear();
+  static sycl::unittest::PiImageArray<2> DevImageArray2[] = {&DevImage1,
+                                                             &DevImage2};
+
+  BundleObj = sycl::compile(KernelBundle);
+  sycl::link(BundleObj);
+  EXPECT_EQ("", current_link_options);
+  EXPECT_EQ(expected_compile_options1, current_compile_options);
+}
+
+TEST(Link_Compile_Options, compile_link_Options_Test_only_link_options) {
```

Comment:
Minor.
I'm not sure it makes a lot of sense to test compile and link options individually. I believe `link_compile_options_all_options` test covers both good enough.

Reply:
I got your point. Now here would be 2 tests: with both empty options and with both options filled.